tests/build/jvm-build.go

```diff
@@ -60,68 +86,84 @@ var _ = framework.JVMBuildSuiteDescribe("JVM Build Service E2E tests", Label("jv
 				}
 				prLog, err := f.TektonController.GetPipelineRunLogs(pr.Name, pr.Namespace)
 				if err != nil {
 					klog.Infof("got error fetching PR logs for %s: %s", pr.Name, err.Error())
 				}
-				klog.Infof("pipeline run log for %s: %s", pr.Name, prLog)
+				filename := fmt.Sprintf("%s-pr-%s.log", pr.Namespace, pr.Name)
+				toDebug[filename] = prLog
+			}
+
+			for _, ab := range abList.Items {
+				v, err := json.MarshalIndent(ab, "", "  ")
+				if err != nil {
+					klog.Infof("error when marshalling content of %s from %s namespace: %+v", ab.Name, ab.Namespace, err)
+				} else {
+					filename := fmt.Sprintf("%s-ab-%s.yaml", ab.Namespace, ab.Name)
+					toDebug[filename] = string(v)
+				}
+			}
+			for _, db := range dbList.Items {
+				v, err := json.MarshalIndent(db, "", "  ")
+				if err != nil {
+					klog.Infof("error when marshalling content of %s from %s namespace: %+v", db.Name, db.Namespace, err)
+				} else {
+					filename := fmt.Sprintf("%s-db-%s.yaml", db.Namespace, db.Name)
+					toDebug[filename] = string(v)
+				}
+			}
+
+			for file, content := range toDebug {
+				if storeLogsInFiles {
+					filename := fmt.Sprintf("%s/%s", testLogsDir, file)
+					if err := os.WriteFile(filename, []byte(content), 0644); err != nil {
+						klog.Infof("cannot write to %s: %+v", filename, err)
+					} else {
+						continue
+					}
+				} else {
+					klog.Infof("%s\n%s", file, content)
+				}
 			}
 		}
-		abList, err := f.JvmbuildserviceController.ListArtifactBuilds(testNamespace)
-		if err != nil {
-			klog.Infof("got error fetching artifactbuilds: %s", err.Error())
-		}
+		// Cleanup
 		for _, ab := range abList.Items {
 			err := f.JvmbuildserviceController.DeleteArtifactBuild(ab.Name, ab.Namespace)
 			if err != nil {
 				klog.Infof("got error deleting AB %s: %s", ab.Name, err.Error())
 			}
 		}
-		dbList, err := f.JvmbuildserviceController.ListDependencyBuilds(testNamespace)
-		if err != nil {
-			klog.Infof("got error fetching dependencybuilds: %s", err.Error())
-		}
 		for _, db := range dbList.Items {
 			err := f.JvmbuildserviceController.DeleteDependencyBuild(db.Name, db.Namespace)
 			if err != nil {
 				klog.Infof("got error deleting DB %s: %s", db.Name, err.Error())
 			}
 		}
-		err = f.TektonController.DeletePipelineRun(prGeneratedName, testNamespace)
-		if err != nil {
-			klog.Infof("error deleting pr %s: %s", prGeneratedName, err.Error())
-		}
-		err = f.TektonController.DeletePipeline("sample-component-build", testNamespace)
-		if err != nil {
-			klog.Infof("error deleting pipeline sample-component-build: %s", err.Error())
-		}
-		err = f.TektonController.DeleteTask("maven", testNamespace)
-		if err != nil {
-			klog.Infof("error deleting task maven: %s", err.Error())
-		}
-		err = f.TektonController.DeleteTask("git-clone", testNamespace)
-		if err != nil {
-			klog.Infof("error deleting task git-clone", err.Error())
-		}
+		Expect(f.HasController.DeleteHasComponent(componentName, testNamespace)).To(Succeed())
```

Comment:
do we really want to fail the tests on cleanup failures ? ... I guess I'm ok with giving this a go, but let's see how often it flakes

Reply:
Regarding cleanup of Component, Application and PipelineRun CRs - we do that in build-service tests as well and I haven't seen a failure in CI related to it yet. So I'd say we're good here. Let's revisit this if we see some issues.

Reply:
ok that's good